src/store/services/user.services.js

```diff
@@ -0,0 +1,83 @@
+import config from 'config';
+// import authHeader from 'store/helpers';
+
+function login(email, password) {
+  const requestOptions = {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify({ email, password })
+  };
+
+  return fetch(`${config.apiUrl}/api/users/login`, requestOptions)
+    .then(handleResponse)
+    .then(user => {
+      // login successful if there's a jwt token in the response
+      if (user.token) {
+        // store user details and jwt token in local storage to keep user logged in between page refreshes
+        localStorage.setItem('user', JSON.stringify(user));
+      }
+
+      return user;
+    });
+}
+
+function logout() {
+  // remove user from local storage to log user out
+  localStorage.removeItem('user');
+}
+
+// function getAll() {
+//   const requestOptions = {
+//     method: 'GET',
+//     headers: authHeader(),
+//   };
+
+//   return fetch(`${config.apiUrl}/users`, requestOptions).then(handleResponse);
+// }
+
+// function getById(id) {
+//   const requestOptions = {
+//     method: 'GET',
+//     headers: authHeader(),
+//   };
+
+//   return fetch(`${config.apiUrl}/users/${id}`, requestOptions).then(handleResponse);
+// }
+
+// function update(user) {
+//   const requestOptions = {
+//     method: 'PUT',
+//     headers: { ...authHeader(), 'Content-Type': 'application/json' },
+//     body: JSON.stringify(user),
+//   };
+
+//   return fetch(`${config.apiUrl}/users/${user.id}`, requestOptions).then(handleResponse);
+// }
+
+function handleResponse(response) {
+  return response.text().then(text => {
+    const data = text && JSON.parse(text);
+    if (!response.ok) {
+      if (response.status === 401) {
+        // auto logout if 401 response returned from api
+        logout();
+        location.reload(true);
+      }
+
+      const error = (data && data.message) || response.statusText;
+      return Promise.reject(error);
```

Comment:
Hi, can you please throw some light on how the method Promise.reject() works in relation to handling responses. Thank You